core/domain/stats_services.py

```diff
@@ -40,6 +40,31 @@ def get_top_unresolved_answers_for_default_rule(exploration_id, state_name):
     }
 
 
+def get_exp_wise_unresolved_answers_count_for_default_rule(exp_ids):
+    exploration_objects = exp_services.get_multiple_explorations_by_id(exp_ids)
+    explorations_ids_list = [
+        exp_domain_objects[1].id
+        for exp_domain_objects in exploration_objects.iteritems()
+        for state_tuple in exp_domain_objects[1].states.iteritems()
+    ]
+    exps_answers_mapping = {}
+
+    for ind, statewise_answers in enumerate(
+            get_top_state_rule_answers_multi(
+                [(exp_domain_objects[1].id, state_tuple[0])
```

Comment:
Actually, if you break this list out, then there is no need for an additional `exploration_ids_list`, since the first element of each tuple in this list corresponds to the exploration id in `exploration_ids_list`. Then, you can also get rid of the additional complexity of having 2 very similar lists representing similar things.